### DISCUSSION TITLE
Publish to pypi

### DISCUSSION THREAD

**Discussion by inverse** [ivandokov/phockup#124] [Ideas]
Publishing this on pypi will enable installation through standard tooling like pip as well as nicer packaging tools such as [pipx](https://pypa.github.io/pipx/).

I recommend using a tool such as [poetry](https://python-poetry.org/) to ease development as well as packaging up the application.

**Comment by ivandokov** [comment]
Thank you for the suggestion. This software is not a Python library that has to live in Python packaging repositories. It just happens to be written in Python and that's completely by accident and my curiosity to try to do something meaningful with the language other than a simple TODO app that will end up in the bin. 

I'd like the software to be packaged in common packages such as snap for Linux, Homebrew for mac and I'm completely open for some automated Windows packaging but that will require GUI so I think a simple one that we tried to create before is a good option. Also I have exactly zero experience developing and shipping Windows apps so PRs are more than welcome in this area!

With that being said I am open for pushing the software to pypi if someone can automate this with Github Actions as I want all deployment actions to be done there.

**Reply by inverse** [reply]
@ivandokov thanks for the reply - that makes total sense. I forgot about the other external things this depends on on which is why docker makes such a nice distribution for this.

**Reply by eoli3n** [reply]
Actually, to me, it makes no sense :)

> This software is not a Python library that has to live in Python packaging repositories.

Who said that pypi only hosts libraries ? There are plenty of software written in python, in pypi.

> I'd like the software to be packaged in common packages such as snap for Linux

Snap is not "for linux" at all. Snap depends on systemd, so it excludes all non-systemd distros, and it's defaulty used by few of systemd distros. It's also full of bugs and have a verrrrrrrry bad design. In you wanted to package for all distros, with that kind of techno, you would move to **flatpaks**. https://www.flatpak.org/

I packaged phockup for void-linux (a non systemd distro) : https://github.com/void-linux/void-packages/pull/39597

For linux:
I strongly suggest to drop snap package for a flatpak

For linux/mac/windows/bsd:
You can use pip on all systems, if you go for a pypi package.

**Reply by ivandokov** [reply]
I am more than happy to accept automated flatpak build & publish PR and I will happily deprecate the snap package!

**Comment by inkhey** [comment]
Hello all, I began working on this problem: https://github.com/ivandokov/phockup/pull/241.
